src/java.base/share/classes/jdk/internal/classfile/impl/ClassPrinterImpl.java

```diff
@@ -0,0 +1,1059 @@
+        return vtis.stream().mapMulti((vti, ret) -> {
+            switch (vti) {
+                case SimpleVerificationTypeInfo s -> {
+                    switch (s) {
+                        case ITEM_DOUBLE -> {
+                            ret.accept("double");
+                            ret.accept("double2");
+                        }
+                        case ITEM_FLOAT ->
+                            ret.accept("float");
+                        case ITEM_INTEGER ->
+                            ret.accept("int");
+                        case ITEM_LONG ->  {
+                            ret.accept("long");
+                            ret.accept("long2");
+                        }
+                        case ITEM_NULL -> ret.accept("null");
+                        case ITEM_TOP -> ret.accept("?");
+                        case ITEM_UNINITIALIZED_THIS -> ret.accept("THIS");
+                    }
+                }
+                case ObjectVerificationTypeInfo o ->
+                    ret.accept(o.className().name().stringValue());
+                case UninitializedVerificationTypeInfo u ->
+                    ret.accept("UNITIALIZED @" + lr.labelToBci(u.newTarget()));
+            }
+        });
+    }
+
+    private record ExceptionHandler(int start, int end, int handler, String catchType) {}
+
+    public static MapNode modelToTree(CompoundElement<?> model, Verbosity verbosity) {
+        return switch(model) {
+            case ClassModel cm -> classToTree(cm, verbosity);
+            case FieldModel fm -> fieldToTree(fm, verbosity);
+            case MethodModel mm -> methodToTree(mm, verbosity);
+            case CodeModel com -> codeToTree((CodeAttribute)com, verbosity);
+        };
+    }
+
+    private static MapNode classToTree(ClassModel clm, Verbosity verbosity) {
+        return new MapNodeImpl(BLOCK, "class")
+                .with(leaf("class name", clm.thisClass().asInternalName()),
+                      leaf("version", clm.majorVersion() + "." + clm.minorVersion()),
+                      list("flags", "flag", clm.flags().flags().stream().map(AccessFlag::name)),
+                      leaf("superclass", clm.superclass().map(ClassEntry::asInternalName).orElse("")),
+                      list("interfaces", "interface", clm.interfaces().stream().map(ClassEntry::asInternalName)),
+                      list("attributes", "attribute", clm.attributes().stream().map(Attribute::attributeName)))
+                .with(constantPoolToTree(clm.constantPool(), verbosity))
+                .with(attributesToTree(clm.attributes(), verbosity))
+                .with(new ListNodeImpl(BLOCK, "fields", clm.fields().stream().map(f ->
+                    fieldToTree(f, verbosity))))
+                .with(new ListNodeImpl(BLOCK, "methods", clm.methods().stream().map(mm ->
+                    (Node)methodToTree(mm, verbosity))));
```

Comment:
```suggestion
                    methodToTree(mm, verbosity))));
```
(Note as i go through the code i am looking at the IDEs report of problems and making note as comments. Likely be more productive for you to do that directly.)